web/src/repo/settings/RepoSettingsCodeIntelligencePage.tsx

```diff
@@ -0,0 +1,125 @@
+                            <small className="text-muted lsif-dump__meta-commit">
+                                <code>{this.props.node.projectRoot.commit.abbreviatedOID}</code>
+                            </small>
+                        </div>
+                    </div>
+
+                    <small className="text-muted lsif-dump__meta-timestamp">
+                        <Timestamp noAbout={true} date={this.props.node.uploadedAt} />
+                    </small>
+                </div>
+            </div>
+        )
+    }
+}
+
+class FilteredLsifDumpsConnection extends FilteredConnection<{}, LsifDumpNodeProps> {}
+
+interface Props extends RouteComponentProps<any> {
+    repo: GQL.IRepository
+}
+
+/**
+ * The repository settings code intelligence page.
+ */
+export const RepoSettingsCodeIntelligencePage: React.FunctionComponent<Props> = ({ repo, ...props }) => {
+    React.useEffect(() => eventLogger.logViewEvent('RepoSettingsCodeIntelligence'), [])
+
+    const dumpsOrError = useObservable(
+        React.useMemo(
+            () =>
+                fetchLsifDumps({ repository: repo.id, isLatestForRepo: true, first: 5000 }).pipe(
+                    map(({ nodes }: { nodes: GQL.ILSIFDump[] }) => sortBy(nodes, node => node.projectRoot.path)),
+                    catchError((error: ErrorLike) => [error])
+                ),
+            [repo.id]
+        )
+    )
+
+    const queryDumps = React.useCallback(
+        (args: FilteredConnectionQueryArgs): Observable<GQL.ILSIFDumpConnection> =>
+            fetchLsifDumps({ repository: repo.id, ...args }),
+        [repo.id]
+    )
+
+    return (
+        <div className="repo-settings-code-intelligence-page">
+            <PageTitle title="Code intelligence" />
+            <h2>Code intelligence</h2>
+            <p>
+                Enable precise code intelligence by{' '}
+                <a href="https://docs.sourcegraph.com/user/code_intelligence/lsif">uploading LSIF data</a>.
+            </p>
+
+            <div className="lsif-dump-collection">
+                <h3>Current LSIF uploads</h3>
+                <p>
+                    These uploads provide code intelligence for the latest commit and are used in cross-repository{' '}
+                    <em>Find Reference</em> requests.
+                </p>
+
+                {dumpsOrError === undefined && <LoadingSpinner className="icon-inline" />}
+                {dumpsOrError !== undefined && isErrorLike(dumpsOrError) && (
+                    <div className="alert alert-danger">
+                        Error getting LSIF uploads for repository:
+                        <br />
+                        <code>{dumpsOrError.message}</code>
+                    </div>
+                )}
+                {dumpsOrError !== undefined && !isErrorLike(dumpsOrError) && dumpsOrError.length > 0 ? (
+                    dumpsOrError.map((dump, i) => <LsifDumpNode key={`latest-${dump.id}`} node={dump} />)
```

Comment:
The `latest-` prefix is not needed, keys only need to be unique within the list they are rendered in (their parent element)
```suggestion
                    dumpsOrError.map(dump => <LsifDumpNode key={dump.id} node={dump} />)
```